### PR TITLE
Memory metrics

### PR DESCRIPTION
**Pull request by ZolotarevIgor-mail-ru**
Part of #124 .
Detailed documentation on memory metrics.

### CODE REVIEW

**Review by vasiliy-t — APPROVED** [review]

File: doc/monitoring/metrics_reference.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+.. _metrics-reference:
+
+===============================================================================
+Metrics reference
+===============================================================================
+
+This page provides detailed description of metrics from ``tarantool.metrics``.
```
Comment (DifferentialOrange) [inline doc/monitoring/metrics_reference.rst:7]: what  `tarantool.metrics` is?

**Review by DifferentialOrange — CHANGES_REQUESTED** [review]

File: doc/monitoring/metrics_reference.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+.. _memory-general:
+
+-------------------------------------------------------------------------------
+Memory general
+-------------------------------------------------------------------------------
+
+Those metrics provide a picture of the whole Tarantool instance.
```
Comment (opomuc) [inline doc/monitoring/metrics_reference.rst:15]: picture of memory usage by Tarantool process.

Comment (opomuc) [inline doc/monitoring/metrics_reference.rst:15]:
```suggestion
Those metrics provide a picture of memory usage by Tarantool process.
```

File: doc/monitoring/metrics_reference.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+.. _memory-allocation:
+
+-------------------------------------------------------------------------------
+Memory allocation
+-------------------------------------------------------------------------------
+
+Provides memory usage report for the slab allocator.
+The slab allocator is the main allocator used to store tuples.
+This can be used to monitor the total memory usage and memory fragmentation.
+To learn more about usage cases, see `this <https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_slab/#box-slab-info>`_
```
Comment (opomuc) [inline doc/monitoring/metrics_reference.rst:44]:
```suggestion
To learn more about use cases, see `this <https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_slab/#box-slab-info>`_
```

File: doc/monitoring/metrics_reference.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+* ``tnt_slab_quota_size`` - the amount of memory available to store tuples and indexes, equals memtx_memory
+
+* ``tnt_slab_arena_size`` - the total memory used for tuples and indexes together (including allocated, but currently free slabs)
+
+* ``tnt_slab_items_size`` - the total amount of memory (including allocated, but currently free slabs) used only for tuples, no indexes
+
+Memory usage, bytes:
+
+* ``tnt_slab_quota_used`` - the amount of memory that is already distributed to the slab allocator
```
Comment (opomuc) [inline doc/monitoring/metrics_reference.rst:56]:
```suggestion
* ``tnt_slab_quota_used`` - the amount of memory that is already reserved by the slab allocator
```

File: doc/monitoring/metrics_reference.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+* ``tnt_slab_items_used_ratio`` - tnt_slab_items_used / tnt_slab_items_size
+
+.. _spaces:
+
+-------------------------------------------------------------------------------
+Spaces
+-------------------------------------------------------------------------------
+
+Those metrics provide information about spaces size.
```
Comment (opomuc) [inline doc/monitoring/metrics_reference.rst:76]:
```suggestion
Those metrics provide specific information about each individual space in Tarantool instance.
```

**Review by opomuc — CHANGES_REQUESTED** [review]

**Review by vasiliy-t — APPROVED** [review]

**Review by opomuc — APPROVED** [review]

**Review by DifferentialOrange — CHANGES_REQUESTED** [review]
In-depth description of default metrics is a really good feature and I am personally requested for it in a couple of tickets. But you have removed more metrics than you have added, and I don't think it's good. Moreover, I'm not sure that the old list of default metrics contained all default metrics enabled by this method.

File: doc/monitoring/metrics_reference.rst [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  bytes in the cache for the tuples stored for the vinyl storage engine.
+
+* ``tnt_info_info_memory_data`` - number of bytes used for storing user data (the tuples)
+  with the memtx engine and with level 0 of the vinyl engine, without taking memory fragmentation into account.
+
+* ``tnt_info_info_memory_index`` - number of bytes used for indexing user data,
+  including memtx and vinyl memory tree extents, the vinyl page index, and the vinyl bloom filters.
+
+* ``tnt_info_info_memory_lua`` - number of bytes used for Lua runtime.
+  Lua memory is bounded by 1-2 GB per instance. Monitoring of this metric can prevent memory overflow.
```
Comment (DifferentialOrange) [inline doc/monitoring/metrics_reference.rst:27]: Isn't it strictly 2 Gb?

**Review by DifferentialOrange — CHANGES_REQUESTED** [review]

**Review by DifferentialOrange — APPROVED** [review]

**Pull request merged**